### PR TITLE
Support DDLs related to distributed schemas from any node - part 1

### PR DESCRIPTION
**Pull request by onurctirtir**
Each commit can be reviewed individually:
* Allow "create / drop schema" from any node for distributed schemas
* Rename Pre/PostprocessAlterDistributedObjectStmt() to Pre/PostprocessAlterDistributedObjectStmtFromCoordinator()
* Allow "alter schema rename to / owner to" from any node for distributed schemas
* Allow supported forms of "create table" variants from any node to create distributed schema tables
* Use IsDroppedOrGenerated() at more places and rename TupleDescColumnNameList()
* Allow "alter table set schema" from any node to move tables into distributed schemas
* Allow "drop table" from any node for distributed schema tables
* Add DisableTrackingQueryCountersForPlannedStmt() for distributed PlannedStmt (prep. for next commit)
* Avoid incrementing query counters when replicating shard placement for alter table set schema
* Allow "alter table" from any node for distributed schema tables
* Add schema_based_sharding_from_workers_a to test distributed schema mgmt commands from any node

I'll keep merging more PRs to the target feature branch for easier reviews.
Subsequent PRs will add;
* More tests especially for "alter table" commands (add / drop column / constraint).
* Support for more DDLs like "create index"

---

At the high level, for the DDLs we want to support from workers, the most significant changes are;
* We call EnsurePropagationToCoordinator() instead of EnsureCoordinator().
* Instead of SendCommandToWorkersWithMetadata(), now we use SendCommandToRemoteNodesWithMetadata() in the code-paths where we need to send a command to other nodes, e.g., to sync metadata changes when creating a distributed schema table from a worker.
* When creating a new distributed schema / schema-distributed table, we need to fetch the next colocation id, shard id and shard placement id. With this PR, whenever we need to consume one of those sequences, we fetch the next sequence value from the coordinator and perform rest of the operations on the current node. This is because, we've never synced those sequences to workers, so we consider their coordinator versions as source of truth.

### CODE REVIEW

File: src/backend/distributed/sql/udfs/citus_drop_trigger/13.1-1.sql [hunk 1]
```diff
@@ -11,9 +11,9 @@ BEGIN
     FOR v_obj IN SELECT * FROM pg_event_trigger_dropped_objects()
                  WHERE object_type IN ('table', 'foreign table')
     LOOP
-        -- first drop the table and metadata on the workers
-        -- then drop all the shards on the workers
-        -- finally remove the pg_dist_partition entry on the coordinator
+        -- first drop the table and metadata on the remote nodes
+        -- then drop all the shards on the remote nodes
+        -- finally remove the pg_dist_partition entry on the local node
```
Comment (onurctirtir) [inline src/backend/distributed/sql/udfs/citus_drop_trigger/13.1-1.sql:16]: Normally, whenever we update the definition of a Citus UDF, we don't modify a SQL file that we created for an older version but we create a new one.

However, since we only update a SQL comment here, I didn't see a need to do it that way.

File: src/backend/distributed/utils/colocation_utils.c [hunk 2]
```diff
@@ -643,7 +649,105 @@ InsertColocationGroupLocally(uint32 colocationId, int shardCount, int replicatio
...
+ * NB: This can be called by any user; for now we have decided that that's
+ * ok. We might want to restrict this to users part of a specific role or such
+ * at some later point.
+ */
+Datum
+citus_internal_get_next_colocation_id(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+
```
Comment (codeforall) [inline src/backend/distributed/utils/colocation_utils.c:741]: Would it make sense to restrict this function so it can only be invoked by Citus‑initiated backends?

**Comment by codeforall** [comment]
The PR is very well written and thoughtfully handles all corner cases. We may need to add a few more complex ALTER TABLE test cases, but that can be addressed in the next phase.
Overall, looks good to go.

**Review by codeforall — APPROVED** [review]

File: src/backend/distributed/utils/colocation_utils.c (hunk 2 above)
Comment (onurctirtir) [reply src/backend/distributed/utils/colocation_utils.c:741]: makes sense 👍
